File: LeetCode/Graph/DFS/JS/Leetcode386-字典序排数.js
var lexicalOrder = function (n) {
    const res = [];
    let dfs = function (cur, n) {
        if (cur <= n) {
            res.push(cur);
        } else {
            return;
        }
        for (let i = 0; i <= 9; i++) dfs(cur * 10 + i, n);
    }
    for (let i = 1; i <= 9; i++) dfs(i, n);
    return res;
};

File: LeetCode/Graph/DFS/Golang/Leetcode386-字典序排数.go
package sbgo

func lexicalOrder(n int) []int {
	var dfs func(int, int)
	res := []int{}
	dfs = func(cur, n int) {
		if cur <= n {
			res = append(res, cur)
		} else {
			return
		}
		for i := 0; i <= 9; i++ {
			dfs(cur*10+i, n)
		}
	}
	for i := 1; i <= 9; i++ {
		dfs(i, n)
	}
	return res
}
